File: game-engine/Domain/Models/ScoutTower.cs
using System;
using System.Collections.Generic;
using Domain.Enums;

namespace Domain.Models
{
    public class ScoutTower : GameObject
    {
        public IList<Guid> Nodes { get; set; }
        public IList<Land> Territory { get; set; }
        public int XRegion { get; set; }
        public int YRegion { get; set; }

        public ScoutTower(Position position) : base(GameObjectType.ScoutTower, position)
        {
            this.Nodes = new List<Guid>();
            this.Territory = new List<Land>();
        }

        public void AddTerritoryNode(Land land)
        {
            Territory.Add(land);
        }
    }
}

File: starter-bots/JavaBot/src/main/java/Models/ScoutTower.java
package Models;

import Enums.GameObjectType;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class ScoutTower extends GameObject {

    List<UUID> nodes;

    public ScoutTower() {
        super(GameObjectType.SCOUT_TOWER);
        this.nodes = new ArrayList<UUID>();
    }

//    public ScoutTower(UUID id, GameObjectType gameObjectType, Position position, List<UUID> nodes) {
//        super(id, gameObjectType, position);
//
//        this.nodes = nodes;
//
//    }

    public List<UUID> getNodes() {
        return nodes;
    }

    public void setNodes(List<UUID> nodes) {
        this.nodes = nodes;
    }
//    @Override
//    public String toString() {
//        return "ScoutTower [   nodes=" + nodes + ", position=" + position  "]";
//    }
}
